### ISSUE TITLE
Glitches when calling restyle from relayout event

### ISSUE THREAD

**Issue by jdugge** [plotly/plotly.js#2384]
When calling `restyle` from a `relayout` event (for example to fetch new data after the axis boundaries have been changed), the trace is sometimes drawn with an offset. The location of the hover info is still in the right place.

![2018-02-20_11-41-49](https://user-images.githubusercontent.com/2399255/36419914-9473b654-1633-11e8-93a1-1ba2ad647ad6.gif)

**Comment by alexcjohnson** [comment]
Interesting... looks to me like this happens when you scroll-zoom, there's enough of a delay during the zoom that we `relayout` but you actually keep going with the scroll-zoom after that, and the delayed `restyle` comes in mid-zoom (for completeness, what actually happens during a zoom is we transform the points during the event, but then once the event is done we remove the transform and reposition the points in their new locations).

If that's what's going on, then the effect could potentially be triggered by any of the zoom/pan interactions, not just scroll-zoom, if a `restyle` occurred during the event; it's just easier to see with scroll-zoom since you're more likely to trigger multiple `relayouts` that way since it's based on a time delay after wheel events, vs other zoom/pan events where the end is a mouseup.

**Comment by gvwilson** [comment]
Hi - this issue has been sitting for a while, so as part of our effort to tidy up our public repositories I'm going to close it. If it's still a concern, we'd be grateful if you could open a new issue (with a short reproducible example if appropriate) so that we can add it to our stack. Cheers - @gvwilson
